NEWS.md

```diff
@@ -96,6 +96,8 @@
 
 14. `setNumericRounding()` now invisibly returns the old rounding value instead of `NULL`, which is now consistent with similar behavior by `setwd()`, `options()`, etc. Thanks @MichaelChirico for the report and @joshhwuu for the fix.
 
+15. `dcast()` now issues a warning when `fun.aggregate` is used but not provided by the user. `fun.aggregate` defaults to `length` in this case; previously, only a message was issued, but often Relying on this default signals unexpected duplicates in the data, so a stricter class of signal was deemed more appropriate, [#5386](https://github.com/Rdatatable/data.table/issues/5386). The warning is classed as `dt_missing_fun_aggregate_warning`, allowing for more targeted handling in user code. Thanks @MichaelChirico for the suggestion and @Nj221102 for the fix.
```

Comment:
> `dcast()` now issues a warning when `fun.aggregate` is used but not provided by the user. `fun.aggregate` defaults to `length` in this case; previously, only a message was issued,

simply informing about the change of message to warning when fun.agg is missing.

> but often Relying on this default signals unexpected duplicates in the data

Updated `but often Relying on this default signals unexpected duplicates in the data` to `However, relying on this default often signals unexpected duplicates in the data.` 

 because "default signals" was a bit confusing even for me,  and this line just informs that when we rely on default ( which is length in this case) means that we have unexpected duplicates in the data, this line was suggested by michael to me so i think this must have some importance, also fixing the typo of making the R capital in this update
 
 > , so a stricter class of signal was deemed more appropriate, The warning is classed as `dt_missing_fun_aggregate_warning`, allowing for more targeted handling in user code.
 
 so we upgraded to warning and warning is classed as `dt_missing_fun_aggregate_warning` for targeted handling using try catch or `globalCallingHandlers`